### PR TITLE
Fix ecbis application configurations

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

Part of the fixes for [#194](https://github.com/opensrp/fhir-resources/issues/194) [#227](https://github.com/opensrp/fhir-resources/issues/227) [#118](https://github.com/opensrp/fhir-resources/issues/118)

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by ellykits — APPROVED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt:133]: can we move this comment to a docstring for the above fun/method?

**Review by pld — COMMENTED** [review]
Wondering, do have a global catch for the "unsupported operation exceptions"?

**Review by pld — CHANGES_REQUESTED** [review]
pls see previous comments

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt
Comment (Rkareko) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt]: This has been updated

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt
Comment (Rkareko) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt]: Removed

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/SearchExtension.kt
Comment (Rkareko) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/SearchExtension.kt]: Support for `Reference` search param type does not exist in main. For this PR we added support for `Date` search param type.  However support for `Reference` search param type can still be added later.

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/SearchExtension.kt]: Cool

**Review by pld — APPROVED** [review]
Lgtm, if there are some other fixes please go ahead, but I don't want to block merge

**Review by pld — APPROVED** [review]

**Pull request merged**
